PYIC-7624: Serve enc key from well-known endpoint

Wynndow opened the pull request:
This configures the private encryption key to be used for decrypting JARs in the CRI stub to be the same for all clients. This means we only need to serve on key on the well-known endpoint.

<!-- Provide a general summary of your changes in the Title above -->
<!-- Include the Jira ticket number in square brackets as prefix, eg `[P4-XXXX] PR Title` -->

## Proposed changes

### What changed

Serve enc key from well-known endpoint

### Why did it change

This configures the private encryption key to be used for decrypting JARs in the CRI stub to be the same for all clients. This means we only need to serve on key on the well-known endpoint.


### Issue tracking
<!-- List any related Jira tickets or GitHub issues -->
<!-- List any related ADRs or RFCs -->
<!-- Delete/copy as appropriate -->

- [PYIC-7624](https://govukverify.atlassian.net/browse/PYIC-7624)


[PYIC-7624]: https://govukverify.atlassian.net/browse/PYIC-7624?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

Wynndow committed:
PYIC-7624: Serve enc key from well-known endpoint

This configures the private encryption key to be used for decrypting
JARs in the CRI stub to be the same for all clients. This means we only
need to serve on key on the well-known endpoint.

Joe-Edwards-GDS reviewed (dismissed):
Seems plausible to me, assuming it won't cause orch to fall over.

The VC signing key will be published on a `did.json` endpoint instead, so we can exclude that for now.

Joe-Edwards-GDS commented:
(once you've fixed the spotless issues I guess!)

Wynndow committed:
PYIC-7624: Defer kid and use to config

Rather than calculating the kid on the fly and setting the use, we
should just use the values that are defined in config. Less suprising.
